widgets/refinement-list.js

```diff
@@ -75,6 +75,13 @@ function refinementList({
   }
 
   return {
+    init: (state, helper) => {
+      // set the maxValuesPerFacet to max(limit, currentValue)
+      var maxValuesPerFacet = helper.getQueryParameter('maxValuesPerFacet');
```

Comment:
Here the widget cannot just override the `maxValuesPerFacet`, it should really set it only if the current one is smaller.